docs/example-react-modal.md

```diff
@@ -4,63 +4,64 @@ title: Modals
 ---
 
 ```jsx
-import React from 'react'
+import React, { useEffect } from 'react'
 import ReactDOM from 'react-dom'
-import { render, fireEvent } from '@testing-library/react'
 
 const modalRoot = document.createElement('div')
 modalRoot.setAttribute('id', 'modal-root')
 document.body.appendChild(modalRoot)
 
-class Modal extends React.Component {
-  el = document.createElement('div')
-  componentDidMount() {
-    modalRoot.appendChild(this.el)
-  }
-  componentWillUnmount() {
-    modalRoot.removeChild(this.el)
-  }
-  render() {
-    return ReactDOM.createPortal(
+const App = ({ onClose, children }) => {
+  const el = document.createElement('div')
+
+  useEffect(() => {
+    modalRoot.appendChild(el)
+
+    return () => modalRoot.removeChild(el)
+  })
+
+  return ReactDOM.createPortal(
+    <div
+      style={{
+        position: 'absolute',
+        top: '0',
+        bottom: '0',
+        left: '0',
+        right: '0',
+        display: 'grid',
+        justifyContent: 'center',
+        alignItems: 'center',
+        backgroundColor: 'rgba(0,0,0,0.3)',
```

Comment:
Since we're tweaking this example, would it make sense to remove stylistic CSS to make it more focused on the problem we're trying to solve?